content/client-lib-development-guide/features.textile

```diff
@@ -642,3 +655,13 @@ h4. CipherParams
 * @(TZ3)@ @Crypto#getDefaultParams@ function
 ** @(TZ3a)@ returns a complete @CipherParams@ instance with randomly generated key and iv
 ** @(TZ3b)@ takes an optional key parameter, returning a @CipherParams@ instance created with that key and with a keyLength calculated from that key
+
+h3(#defaults). Client Library defaults
+
+The following default values are configured for the client library:
+
+* @(DF1)@ Realtime defaults:
+** @(DF1a)@ @connectionStateTtl@ integer - default 60s. The duration that Ably will persist the connection state when a Realtime client is abruptly disconnected. When the client is in the @DISCONNECTED@ state, once this TTL has passed, the client should change the state to the @SUSPENDED@ state signifying that the state is now lost i.e. channels need to be reattached manually.
+Note that this default is override by any @connectionStateTtl@ specified in the @ConnectionDetails@ of the @CONNECTED@ @ProtocolMessage@
+** @(DF1b)@ @realtimeCloseTimeout@ - default 10s. Following an explicit request to close a connection from the client library by sending a @CLOSE@ @ProtocolMessage@, this value is the amount of time that the client library will wait before forcibly closing the connection and moving the client's connection state to @CLOSED@
```

Comment:
So you want a generic timeout that applies to any ProtocolMessage sent that expects a response, and thus assumes the request has failed?  I am not sure I would necessarily agree, assuming I sent a message, then the connection dropped, and was resumed 20s later, I would not expect an automatic timeout on the Message publish as it may have been sent before the connection disconnected briefly.